pkg/apis/crd/v1alpha1/types.go

```diff
@@ -427,3 +354,132 @@ type BGPPeer struct {
 	// a restart before deleting stale routes. The range of the value is from 1 to 3600, and the default value is 120.
 	GracefulRestartTimeSeconds *int32 `json:"gracefulRestartTimeSeconds,omitempty"`
 }
+
+// Source describes the source spec of the packetcapture.
+type Source struct {
+	// Namespace is the source namespace.
+	Namespace string `json:"namespace"`
+	// Pod is the source pod.
+	Pod string `json:"pod"`
+	// IP is the source IPv4 or IPv6 address.
+	IP *string `json:"ip,omitempty"`
```

Comment:
Is this field mutually exclusive with `Namespace` / `Pod`?

Reply:
not exactly. we don't have this kind of check now. This `IP` field is used for  `Pod->IP` or `IP->Pod` filter case.

The only requirement is that at least one pod is specified.

Reply:
So it would be valid to provide both a destination Pod and a destination IP (which don't match)?

Reply:
As long as there is one pod present(no matter whether it's in the source or destination), it will work(ignore the ip field in the same endpoint). I didn't add strict validation for the IP related case, not sure what's the best approach.

Reply:
since it's exclusive , i have added crd validation using `oneOf` in the yaml. User cannot provide `ip` and `pod` both in the source/dest struct in there CR.

Reply:
@antoninbas found a little problem with this whole `omitempty` thing.

https://github.com/antrea-io/antrea/pull/5821/commits/4ef7a4b15336adb6307e3d952756f4ee9676c25f

Since i'm using `oneOf` to perform some validation in the crd,  so the related field need to be set to `omitempty` wether they are pointer or value or not , or the validation will fail. (using yaml is fine, but golang struct is not, eg in tests code)

For example. If i want to use oneOf for source.ip/pod, means only one can be spcified. the following wont work

```golang
type Source struct {
	Namespace string `json:"namespace"`
	Pod string `json:"pod"`
	IP string `json:"ip"`
}
```
```golang
spec: crdv1alpha1.PacketCaptureSpec{
					Source: crdv1alpha1.Source{
						Namespace: "test-namespace",
						Pod:       "test-pod",
					},
```

since there is no `omitempty` it's actually:
```golang
spec: crdv1alpha1.PacketCaptureSpec{
					Source: crdv1alpha1.Source{
						Namespace: "test-namespace",
						Pod:       "test-pod",
                                                IP:         "",
					},
```

So the `oneOf` validation will fail. The solution would be change every affected field (including `pod`)  to pointer + omitempty. or keep as value + omitempty. If not set, it won't show up. 

Currently i kept the value format and add some of the `omitempty` tag back, as change to pointer would require some changes on the current codebase. If you still felt this is needed ( > pointer), i can update this later.

Reply:
well I think that in K8s this would be something like:
```go
type Source {
    SourcePod *PodReference `json:"sourcePod,omitempty"`
    SourceIP *string `json:"sourceIP,omitempty"`
}

type Destination {
    DestinationPod *PodReference `json:"destinationPod,omitempty"`
    DestinationService *ServiceReference `json:"destinationService,omitempty"`
    DestinationIP *string `json:"destinationIP,omitempty"`
}
```

Technically the pointers are not strictly necessary for the IP fields, but I like the symmetry.

It does make the API more verbose though, cc @tnqn

Reply:
now that we have addressed the issue, could you put back the `oneOf`s in the CRD OpenAPI schema?

Reply:
sure ,already did.